Client/Frontend/Browser/BrowserViewController.swift

```diff
@@ -612,45 +611,44 @@ class BrowserViewController: UIViewController {
         return false
     }
 
-    override func observeValueForKeyPath(keyPath: String, ofObject object: AnyObject, change: [NSObject: AnyObject], context: UnsafeMutablePointer<Void>) {
+    override func observeValueForKeyPath(keyPath: String?, ofObject object: AnyObject?, change: [String: AnyObject]?, context: UnsafeMutablePointer<Void>) {
         let webView = object as! WKWebView
         if webView !== tabManager.selectedTab?.webView {
             return
         }
-
-        switch keyPath {
+        guard let path = keyPath else { assertionFailure("Unhandled KVO key: \(keyPath)"); return }
+        switch path {
         case KVOEstimatedProgress:
-            let progress = change[NSKeyValueChangeNewKey] as! Float
+            guard let progress = change?[NSKeyValueChangeNewKey] as? Float else { break }
             urlBar.updateProgressBar(progress)
             // when loading is stopped, KVOLoading is fired first, and only then KVOEstimatedProgress with progress 1.0 which would leave the progress bar running
             if progress != 1.0 || tabManager.selectedTab?.loading ?? false {
                 auralProgress.progress = Double(progress)
             }
         case KVOLoading:
-            let loading = change[NSKeyValueChangeNewKey] as! Bool
+            guard let loading = change?[NSKeyValueChangeNewKey] as? Bool else { break }
             toolbar?.updateReloadStatus(loading)
             urlBar.updateReloadStatus(loading)
             auralProgress.progress = loading ? 0 : nil
         case KVOURL:
-            if let tab = tabManager.selectedTab where tab.webView === webView {
-                updateURLBarDisplayURL(tab)
+            guard let tab = tabManager.selectedTab where tab.webView === webView else { break }
+            updateURLBarDisplayURL(tab)
 
-                scrollController.showToolbars(animated: false)
+            scrollController.showToolbars(animated: false)
 
-                if let url = tab.url {
-                    if ReaderModeUtils.isReaderModeURL(url) {
-                        showReaderModeBar(animated: false)
-                    } else {
-                        hideReaderModeBar(animated: false)
-                    }
+            if let url = tab.url {
+                if ReaderModeUtils.isReaderModeURL(url) {
+                    showReaderModeBar(animated: false)
+                } else {
+                    hideReaderModeBar(animated: false)
                 }
-                updateInContentHomePanel(tab.url)
             }
+            updateInContentHomePanel(tab.url)
```

Comment:
I think we want to keep this line inside the above if statement. Don't see why we want to move it outside.